### PR TITLE
Add Ibis server

### PR DESCRIPTION
**Pull request by grieve54706**
Add a python server for [ibis-project/ibis](https://github.com/ibis-project/ibis)

# Query data API

```sql
POST /v2/ibis/{data_source}/query
```

## **Path parameters**

- data_source: string, `required`. The name of the data source to query.
    
    The available values*:*
    
    - postgres
    - bigquery
    - snowflake

## Request body

- sql: string, `required`

Other fields determined by data source

For Postgres:

- host: string, `required`
- port: integer, `required`
- user: string, `required`
- password: string, `required`

```json
{
  "sql": "select * from orders limit 5",
  "host": "localhost",
  "port": 5432,
  "user": "postgres",
  "password": "postgres"
}
```

For Bigquery:

- project_id: string, `required`
- dataset_id: string, `required`
- credentials: Base64 encode `credentials.json`

```json
{
  "sql": "select * from tpch_tiny.orders limit 5",
  "project_id": "wrenai",
  "dataset_id": "tpch_tiny",
  "credentials": "ewogICJ0eXBlIjogInNlcnZpY2VfYWNjb3VudCIsCiAgInByb2plY3Rfa..."
}
```

For Snowflake:

- user: string, `required`
- password: string, `required`
- account: string, `required`
- database: string, `required`
- schema: string

```json
{
  "sql": "select * from TPCH_SF1.ORDERS LIMIT 5",
  "user": "user",
  "password": "password",
  "account": "snowflake_account",
  "database": "SNOWFLAKE_SAMPLE_DATA"
}
```

## Response body

- columns
- data
- dtypes: Type through the pandas
```json
{
    "columns": [
        "o_orderkey",
        "o_custkey",
        "o_orderstatus",
        "o_totalprice",
        "o_orderdate",
        "o_orderpriority",
        "o_clerk",
        "o_shippriority",
        "o_comment"
    ],
    "data": [
        [
            1,
            370,
            "O",
            172799.49,
            820540800000,
            "5-LOW",
            "Clerk#000000951",
            0,
            "nstructions sleep furiously among "
        ]
    ],
    "dtypes": {
        "o_orderkey": "int32",
        "o_custkey": "int32",
        "o_orderstatus": "object",
        "o_totalprice": "object",
        "o_orderdate": "object",
        "o_orderpriority": "object",
        "o_clerk": "object",
        "o_shippriority": "int32",
        "o_comment": "object"
    }
}
```

### CODE REVIEW

File: ibis-server/dto.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    pass
+
+
+class PostgresDTO(IbisDTO):
+    host: str = Field(examples=["localhost"])
+    port: int = Field(default=5432)
+    user: str
+    password: str
+
+
```
Comment (goldmedal) [inline ibis-server/dto.py:15]: Just take a note:
I notices that there're different number of empty line between functions or classes. I think we may can find some solution to format the code. To ensure everyone has same coding style.

**Review by goldmedal — APPROVED** [review]
Nice work. LGTM.

File: ibis-server/dto.py (hunk 1 above)
Comment (cyyeh) [reply ibis-server/dto.py:15]: @goldmedal @grieve54706 I recommend using ruff. https://github.com/astral-sh/ruff

Comment (grieve54706) [reply ibis-server/dto.py:15]: My default IDE follows the PEP8. It's a regular coding style.

Comment (grieve54706) [reply ibis-server/dto.py:15]: > @goldmedal @grieve54706 I recommend using ruff. https://github.com/astral-sh/ruff

Nice idea, I will add it in the future.

**Pull request merged**